demo/streaming/streaming.hs

```diff
@@ -6,7 +7,9 @@
 import Data.Conduit
 import qualified Data.Conduit.Binary as CB
 import Control.Concurrent.Lifted (threadDelay)
-import Data.Monoid ((<>))
+#if !MIN_VERSION_base(4,11,0)
+import Data.Semigroup ((<>))
+#endif
```

Comment:
`base-4.11` corresponds to GHC 8.4. We currently have a `base` lower bound on `4.10` for GHC 8.2. 8.4 was released in March 2018, almost 7 years ago, well outside of any reasonable need to preserve support.

Honestly I'm in favor of bumping the `base` lower bound and deleting all this CPP.

Reply:
@parsonsmatt 
Should I also bump the (patch) version of every package I set to `base >= 4.11`? (And add to the Changelog?)